client/src/components/MapView.vue

```diff
@@ -208,6 +207,13 @@ export default defineComponent({
 
       this.setUpInteractionHandlers();
       this.setUpControls();
+
+      // Check whether there's a layer selected in the router.
+      if (router.currentRoute.value.query?.layer != null) {
+        this.state.currentDataLayer = this.state.dataLayers.find(l => l.styleLayer.id == router.currentRoute.value.query.layer) ?? leadServiceLinesByWaterSystemLayer;
```

Comment:
nit: layer instead of 'l' - it looks like a 1. Also can we not just check whether the query param is present? if the param value is invalid shouldn't we remove it instead of ignoring it?

Reply:
You point out something good because I realize I can just do 
this.state.setCurrentDataLayer(this.state.dataLayers.find(layer => layer.styleLayer.id == router.currentRoute.value.query?.layer)
        ?? leadServiceLinesByWaterSystemLayer);

Now if there is a misspelling in the routing it just redirects to the water systems data layer